Quick start updates, Example updates, and Updates to SBO topics for modular-docs template and content type attributes alignment (#41809)
openshift/openshift-docs: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service.adoc

. Create a `ServiceBinding` CR to project the binding data:
+

> ----

Reviewer: I noticed that the openshift-docs use `[source,terminal]` consistently above command examples. How does using that line affect the appearance of the command in the published documentation?
```suggestion
[source,terminal]
----
```

Reviewer: Would it help to use it consistently with the commands and output examples in this PR?

Reviewer: See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

> $ oc apply -n my-petclinic -f - << EOD

Reviewer 2: @rolfedh got it. Thanks for the clarification

The output verifies that the `ServiceBinding` CR is created to project the binding data into the sample application.
+
.Example output
[source,terminal]
----
servicebinding.binding.operators.coreos.com/spring-petclinic created
----

. Verify that the request for service binding is successful:
+
[source,terminal]
----
$ oc get servicebindings -n my-petclinic
----
+
.Example output
[source,terminal]
----
NAME                          READY   REASON              AGE
spring-petclinic-postgresql   True    ApplicationsBound   47m
----
+
By default, the values from the binding data of the database service are projected as files into the workload container that runs the sample application. For example, all the values from the Secret resource are projected into the `bindings/spring-petclinic-pgcluster` directory.

. Verify that the files in the application contain the projected binding data, by printing out the directory contents:
+
[source,terminal]
----

> $ for i in username password host port type; do oc exec -it deploy/spring-petclinic -n my-petclinic -- /bin/bash -c 'find /bindings/*/'$i' -exec echo -n {}:" " \; -exec cat {} \;'; echo; done

Reviewer 2: @Srivaralakshmi If you want to keep this verification step in the quickstart, the command needs to be slightly modified:

```suggestion
$ for i in username password host port type; do oc exec -it deploy/spring-petclinic -n my-petclinic -- /bin/sh -c 'find /bindings/*/'$i' -exec echo -n {}:" " \; -exec cat {} \;'; echo; done
```

Author: @pmacik In that case, I am removing this step.

----
+
.Example output: With all the values from the secret resource
[source,text]
----
/bindings/spring-petclinic/username: hippo
/bindings/spring-petclinic/password: w0ZB<0j1W|K;+4*TlK7-w^z/
/bindings/spring-petclinic/host: hippo-pgbouncer
/bindings/spring-petclinic/port: 5432
/bindings/spring-petclinic/type: postgresql

> ----

Author: @pmacik: Is this step really required?  What is the value add of this step? If there is no specific purpose/value add, can we remove this step? Does it bring any significant difference to the user experience?

Author: @pmacik: If the content is good and there are no further comments, can you please approve? Thanks!

Reviewer 2: @Srivaralakshmi This is definitelly not required or necessary. Originally, this was included as per your suggestion to have a verification that the projection of the binding data was successfull, nothing more.

The output verifies that the `ServiceBinding` CR is created to project the binding data into the sample application.
+
.Example output
[source,terminal]
----
servicebinding.binding.operators.coreos.com/spring-petclinic created
----

. Verify that the request for service binding is successful:
+
[source,terminal]
----
$ oc get servicebindings -n my-petclinic
----
+
.Example output
[source,terminal]
----
NAME                          READY   REASON              AGE

> spring-petclinic-postgresql   True    ApplicationsBound   47m

Reviewer 2: ```suggestion
spring-petclinic-pgcluster   True    ApplicationsBound   47m
```
The name of the `ServiceBinding` resource is `spring-petclinic-pgcluster`

Author: Done.

----
+
By default, the values from the binding data of the database service are projected as files into the workload container that runs the sample application. For example, all the values from the Secret resource are projected into the `bindings/spring-petclinic-pgcluster` directory.
+
[NOTE]
====
Optionally, you can also verify that the files in the application contain the projected binding data, by printing out the directory contents:

[source,terminal]
----
$ for i in username password host port type; do oc exec -it deploy/spring-petclinic -n my-petclinic -- /bin/bash -c 'cd /tmp; find /bindings/*/'$i' -exec echo -n {}:" " \; -exec cat {} \;'; echo; done
----

.Example output: With all the values from the secret resource
[source,text]
----

> /bindings/spring-petclinic/username: hippo
> /bindings/spring-petclinic/password: w0ZB<0j1W|K;+4*TlK7-w^z/
> /bindings/spring-petclinic/host: hippo-primary.my-petclinic.svc
> /bindings/spring-petclinic/port: 5432
> /bindings/spring-petclinic/type: postgresql

Reviewer 2: ```suggestion
/bindings/spring-petclinic-pgcluster/username: hippo
/bindings/spring-petclinic-pgcluster/password: w0ZB<0j1W|K;+4*TlK7-w^z/
/bindings/spring-petclinic-pgcluster/host: hippo-primary.my-petclinic.svc
/bindings/spring-petclinic-pgcluster/port: 5432
/bindings/spring-petclinic-pgcluster/type: postgresql
```
The name of the binding directory should match the name of the `ServiceBinding` resource

Reviewer 2: @Srivaralakshmi ^^

Author: Done.

+

> .Example output

Reviewer: ```suggestion
.Example output
[source,terminal]
```

----
servicebinding.binding.operators.coreos.com/spring-petclinic created
----

> . Verify that the request for service binding is successful:

Reviewer: How does a user know that the request was successful? What should the servicebindings status be?

> . Verify that the request for service binding is successful:

Reviewer: Should verification steps like this on have their own separate verification section, [like the one shown in the template](https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc)?
```suggestion
.Verification

. Verify that the request for service binding is successful:
```

Author: Thanks, @rolfedh for the suggestion. Ideally yes, but this QSG has multiple procedures and some have verification steps for each step, so to maintain consistency I prefer not to use a separate Verification section for now, but to keep it aligned with every step as required.

+

> ----

Reviewer: ```suggestion
[source,terminal]
----
```

Reviewer: etc...

$ oc get servicebindings -n my-petclinic
----
+
.Example output
----
NAME                          READY   REASON              AGE
spring-petclinic-postgresql   True    ApplicationsBound   47m
----
+
By default, the values from the binding data of the database service are projected as files into the workload container that runs the sample application. For example, all the values from the Secret resource are projected into the `bindings/spring-petclinic-pgcluster` directory.

> . Verify that the files in the application contain the projected binding data, by using the following command in shell to print out the contents:

Reviewer: ```suggestion
. Verify that the files in the application contain the projected binding data. Use the following command in shell to print out the contents:
```

Reviewer: Many instructions in the openshift-docs omit the "using the following command in shell to print" language. They give the instruction on one line and then the command on the next. You can choose either approach.

> . Verify that the files in the application contain the projected binding data, by using the following command in shell to print out the contents:

Reviewer: For consistency and minimalism, please follow the [Code blocks, command syntax, and example output](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output) section of the OpenShift Documentation guidelines to these commands.

> . Verify that the files in the application contain the projected binding data, by using the following command in shell to print out the contents:

Reviewer: ```suggestion
. To verify that the files in the application contain the projected binding data, enter the following command:

```

Reviewer: Also use `[source,terminal]`

Author: @rolfedh I prefer this suggestion and have implemented it already: https://github.com/openshift/openshift-docs/pull/41809/#discussion_r805801585

+
----

> $ for i in username password host port type; do oc exec -it deploy/spring-petclinic -n my-petclinic -- /bin/bash -c 'find /bindings/*/'$i' -exec echo -n {}:" " \; -exec cat {} \;'; echo; done

Reviewer: Check the rendered output topic for this file to make sure the italics that start in this command don't affect the following steps.

Reviewer: Also use `[source,terminal]`

----
+
.Example output: With all the values from the secret resource
----
/bindings/spring-petclinic/username: hippo
/bindings/spring-petclinic/password: w0ZB<0j1W|K;+4*TlK7-w^z/
/bindings/spring-petclinic/host: hippo-pgbouncer
/bindings/spring-petclinic/port: 5432
/bindings/spring-petclinic/type: postgresql
----

. Set up the port forwarding from the application port to access the sample application from your local environment:
+
----
$ oc port-forward --address 0.0.0.0 svc/spring-petclinic 8080:80 -n my-petclinic
----
+
.Example output
----
Forwarding from 0.0.0.0:8080 -> 8080
Handling connection for 8080
----

> 

Reviewer: Would it make sense to combine the following lines into a single step and put them in a new `.Verification` section?
